src/ducks/apps/components/Sections/Sections.jsx

```diff
@@ -85,9 +85,12 @@ export class Sections extends Component {
     super(props, context)
     this.state = {
       filter: {},
-      searchFieldValue: '',
-      filteredApps: this.getFilteredApps()
+      searchFieldValue: ''
     }
+
+    // getFilteredApps must have state.filter already ready
+    this.state.filteredApps = this.getFilteredApps()
```

Comment:
```suggestion
this.state = { 
filter: {},
searchFieldValue: '',
// getFilteredApps must have state.filter already ready
filteredApps: this.getFilteredApps()
}
```

I found it clearer

Reply:
This cannot work as `this.state` is declared after `this.getFilteredApps()` runs but `this.getFilteredApps` relies on `this.state`. This is why there is a comment. I could replace "already ready" by "already declared", I think this would make it clearer.

Reply:
Also, you can't set this `filteredApps` on did/WillMount() for instance?

Reply:
> getFilteredApps can also does something

Yes it can. I envisioned this possibility and prefer the other way.

> Also, you can't set this filteredApps on did/WillMount() for instance?

I don't understand what you mean to say by that ? Do I need to do something ?

Reply:
Instead of calling getFilteredApps in the constructor and dealing with the fact that state.filter should have been instanciated, you can call getFilteredApps in the willmount. Like that, you're sure that state.filter is already declared

Reply:
I am not sure what problem we are solving here ? `componentWillMount` is deprecated

I do no think that
```
componentWillMount() {
  this.setState({ filteredApps: this.getFilteredApps() })
}
```

is better than the original code.

Reply:
didMount does the trick without using deprecated method. 

By doing so, we solve the issue when reading this code, not understanding why `state.filteredApps` is managed differently than the others and having to read the comment and the `getFilteredApps()` code to understand why. 

But never mind

Reply:
Yep, but doing it in didMount will incure another render, this is why I had done in the constructor and I will change the comment so that it's clearer.